src/mlpack/methods/ann/init_rules/he_init.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file he_init.hpp
+ * @author Dakshit Agrawal
+ *
+ * Intialization rule given by He et. al. for neural networks. The He
+ * initialization initializes weights of the neural network to better
+ * suit the rectified activation units.
+ *
+ * For more information, the following paper can be referred to:
+ *
+ * @code
+ * @article{He2015DelvingDI,
+ * title={Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification},
+ * author={Kaiming He and Xiangyu Zhang and Shaoqing Ren and Jian Sun},
+ * journal={2015 IEEE International Conference on Computer Vision (ICCV)},
+ * year={2015},
+ * pages={1026-1034}}
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_INIT_RULES_HE_INIT_HPP
+#define MLPACK_METHODS_ANN_INIT_RULES_HE_INIT_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * This class is used to initialize weight matrix with the He initialization rule.
+ */
+class HeInitialization
+{
+ public:
+    /**
+     * Initialize the HeInitialization object.
+     *
+     */
+    HeInitialization()
+    {
+        // Nothing to do here.
+    }
+
+    /**
+     * Initialize the elements of the weight matrix with the He initialization
+     * rule.
+     *
+     * @param W Weight matrix to initialize.
+     * @param rows Number of rows.
+     * @param cols Number of columns.
+     */
+    void Initialize(arma::mat& W,
+                    const size_t rows,
+                    const size_t cols)
+    {
+        // He initialization rule says to initialize weights with random
+        // values taken from a gaussian distribution with mean = 0 and
+        // standard deviation = sqrt(2/rows), i.e. variance = (2/rows).
+        double_t variance =  2.0 / ((double) rows);
```

Comment:
Why double_t and not double?

Reply:
There is no particular reason.  But on searching, I found that it is better to use double_t to make the code more adaptive to compile on both Windows and Unix.  However, it shouldn't make much difference.  Do let me know which one to use.

Reply:
>  But on searching, I found that it is better to use double_t to make the code more adaptive to compile on both Windows and Unix.

Can you dig into that and explain a little of what you found please?

In any case we should stick with `double` since it's what's used everywhere else.

Reply:
I am sorry, but I could not find any concrete reason for using `double_t` over `double`.  I have changed `double_t` to `double` in the code.